### PR TITLE
getting-started: add host port forwarding to QEMU example

### PR DESCRIPTION
**Pull request by rfairley**
See details in commit message.

---

Closes: https://github.com/coreos/fedora-coreos-docs/issues/6

### CODE REVIEW

**Comment by rfairley** [comment]
ref #63 for when rebasing, in case this gets merged beforehand

File: modules/ROOT/pages/getting-started.adoc [hunk 1]
```diff
@@ -48,28 +49,32 @@ qemu-kvm -m 2048 -cpu host -nographic -snapshot \
...
 ----
 +
 .Example launching FCOS with virt-install
 [source, bash]
 ----
 virt-install --connect qemu:///system -n fcos -r 2048 --os-variant=fedora31 --import \
 	--graphics=none --disk size=10,backing_store=/path/to/fedora-coreos-qemu.qcow2 \
-	--qemu-commandline="-fw_cfg name=opt/com.coreos/config,file=/path/to/example.ign"
+	--qemu-commandline="-fw_cfg name=opt/com.coreos/config,file=/path/to/example.ign" \
+	--qemu-commandline="-device virtio-net,netdev=user.0 -netdev user,id=user.0,hostfwd=tcp::2222-:22"
```
Comment (jlebon) [inline modules/ROOT/pages/getting-started.adoc:62]: By default, `virt-install` should already be attaching the default network. Is that not the case on your box? But more generally, I think we should let libvirt be aware of the network config here instead of passing it "behind its back" via `--qemu-commandline`.

File: modules/ROOT/pages/getting-started.adoc [hunk 2]
```diff
@@ -48,28 +49,32 @@ qemu-kvm -m 2048 -cpu host -nographic -snapshot \
...
 ----
 
 NOTE: When using `virt-install`, you must specify absolute paths to the image and the Ignition file.
 
 TIP: If running with SELinux enabled, you may need to change the label of the Ignition file to allow access: `chcon -t svirt_home_t path/to/example.ign`
 
-Once the VM has finished booting, its IP addresses will appear on the serial console. You can then SSH into the FCOS VM as the `core` user:
+Once the VM has finished booting, its IP addresses will appear on the serial console. You can then SSH into the FCOS VM as the `core` user (replacing `localhost` and `2222` with the IP address and port of the VM, if different values from the suggested values mentioned earlier were chosen):
```
Comment (jlebon) [inline modules/ROOT/pages/getting-started.adoc:69]: Once we simplify the `virt-install` one, I think we can reword this to:

> (on QEMU, use `localhost` and pass `-p 2222` or whichever port you've forwarded)

or something like that?

File: modules/ROOT/pages/getting-started.adoc [hunk 3]
```diff
@@ -48,28 +49,32 @@ qemu-kvm -m 2048 -cpu host -nographic -snapshot \
...
 
 [source, bash]
 ----
-ssh core@<ip address>
+ssh core@localhost -p 2222
```
Comment (jlebon) [inline modules/ROOT/pages/getting-started.adoc:73]: I'd keep this unchanged since it should work as is for libvirt, and for QEMU we'll have the notice from the previous paragraph.

File: modules/ROOT/pages/getting-started.adoc [hunk 4]
```diff
@@ -48,28 +49,32 @@ qemu-kvm -m 2048 -cpu host -nographic -snapshot \
...
 ----
 
+NOTE: With QEMU/libvirt, the IP addresses are assigned by the QEMU usermode networking backend, and will not be directly visible from your host machine. However, via host port forwarding through `localhost` (shown above), the machine is accessible.
```
Comment (jlebon) [inline modules/ROOT/pages/getting-started.adoc:76]: Hmm, feels like we can drop this paragraph given that it's explained in the sentence just before the `qemu-kvm` command-line?

File: modules/ROOT/pages/getting-started.adoc (hunk 3 above)
Comment (lucab) [reply modules/ROOT/pages/getting-started.adoc:73]: Better yet, move to a `${LOCALPORT}` variable which is `2222` on qemu and `22` on libvirt.

File: modules/ROOT/pages/getting-started.adoc (hunk 4 above)
Comment (lucab) [reply modules/ROOT/pages/getting-started.adoc:76]: Though I think we should have a note here to mention that the address shown in the console in qemu is not reachable from the host, and SSH port is forwarded to an higher port on localhost instead.

File: modules/ROOT/pages/getting-started.adoc (hunk 1 above)
Comment (rfairley) [reply modules/ROOT/pages/getting-started.adoc]: Ahh, I had misunderstood how `virt-install` set up the network previously, thinking it used a similar usermode networking setup like QEMU. Makes a lot more sense to keep the networking set up via libvirt, and just use the IP address shown on the FCOS serial console.

File: modules/ROOT/pages/getting-started.adoc (hunk 2 above)
Comment (rfairley) [reply modules/ROOT/pages/getting-started.adoc]: SGTM, added!

File: modules/ROOT/pages/getting-started.adoc (hunk 3 above)
Comment (rfairley) [reply modules/ROOT/pages/getting-started.adoc]: :+1: to keeping this unchanged, and have the note above in https://github.com/coreos/fedora-coreos-docs/pull/86#discussion_r443684364. `${LOCALPORT}` sounds good, but  if we did this I think it'd look neater to have an `${IPADDRESS}` variable too (choosing localhost or the one assigned by libvirt), IMHO, to avoid a variable substitution in one part of the line but deleting/overwriting `<ip address>` in another. Since the user would need to enter the IP address manually anyway in the unchanged version or with a variable, and it wouldn't be too much more typing for `-p 2222`, I've left this as it is. Though really this is just minor preference, and I'd still be happy using the variables mentioned with a nudge (either both the IPADDRESS and LOCALPORT or just LOCALPORT).

File: modules/ROOT/pages/getting-started.adoc (hunk 4 above)
Comment (rfairley) [reply modules/ROOT/pages/getting-started.adoc]: :+1: I condensed this note a bit, and kept the explanation at the above explanation (mentioning briefly the IP address not being reachable).

**Comment by jlebon** [comment]
Thanks for this!

**Comment by rfairley** [comment]
Updated - good to merge is all is good!

**Comment by rfairley** [comment]
Thanks for the reviews!
